File: RoRServer/src/main/java/models/dataTranserObject/MessageInformation.java
package models.dataTranserObject;

import java.util.HashMap;
import java.util.Map;

public class MessageInformation {
    private String clientId;
    private Map<String, Object> attributes = new HashMap<String, Object>();

    public MessageInformation(){}

    public MessageInformation(String clientid, HashMap<String, Object> attributes) {
        this.clientId = clientid;
        this.attributes = attributes;
    }

    public String getClientid() {
        return clientId;
    }

    public String toString() {
        return String.format("client ID: {0}, \nattributes Map: {2}", clientId, attributes.toString());
    }

    public void setClientid(String clientId) {
        this.clientId = clientId;
    }

    public void putValue(String key, Object value) {
    	attributes.put(key, value);
    }
    
    public String getValueAsString(String key) {
    	return (String)attributes.get(key);
    }
    
    public int getValueAsInt(String key) {
    	return (int)attributes.get(key);
    }
    
    public double getValueAsDouble(String key) {
    	return (double)attributes.get(key);
    }
    
    public boolean getValueAsBoolean(String key) {
    	return (boolean)attributes.get(key);
    }
}

File: RoRClient/Model/DataTransferObject/MessageInformation.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using RoRClient.Model.Models;
using Newtonsoft.Json.Linq;
using RoRClient.Model.Connections;

namespace RoRClient.Model.DataTransferObject
{
    //Klasse für den Inhalt der Messages an den Server (folgt dem Schema: ClientId, Request, Attributes)
    public class MessageInformation
    {
        public readonly String clientId;
        public Dictionary<string, Object> attributes;

        public Dictionary<string, Object> Attributes
        {
            get
            {
                return attributes;
            }
        }

        public MessageInformation()
        {
            clientId = ClientConnection.GetInstance().ClientId.ToString();
            attributes = new Dictionary<string, object>();
        }

        public void PutValue(string key, object value)
        {
            attributes.Add(key, value);
        }
        
        public String GetValueAsString(string key)
        {
            return (string)attributes[key];
        }

        public int GetValueAsInt(string key)
        {
            object obj = attributes[key];
            if(obj.GetType() == typeof(double))
            {
                return Convert.ToInt32(obj);
            }
            string temp = obj.GetType().ToString();
            return Convert.ToInt32(obj);
        }

        public double GetValueAsDouble(string key)
        {
            return (double)attributes[key];
        }

        public bool GetValueAsBool(string key)
        {
            return (bool)attributes[key];
        }

        public JObject GetValueAsJsonObject(string key)
        {
            return JObject.Parse(GetValueAsString(key));
        }
    }
}
